spec/openapi3.yaml

```diff
@@ -2805,6 +2835,55 @@ components:
       allOf:
         - $ref: "#/components/schemas/EmbeddedTransactionDTO"
         - $ref: "#/components/schemas/TransferTransactionBodyDTO"
+
+    # Schemas - Metadata
+    KeyMetadataDTO:
+      type: object
+      required:
+      - key
+      - metadata
+      properties:
+        key:
+          type: string
+          format: hex
+          description: Metadata key.
+          example: EB6839C7E6BD0031FDD5F8CB5137E9BC950D7EE7756C46B767E68F3F58C24390
+        metadata:
+          type: array
+          description: Array of metadata objects.
+          items:
+            anyOf:
```

Comment:
If you decide to return the same object always, you could remove the ``anyOf`` property and keep just one "MetadataDTO". 

However, it seems that the Metadata is an object is going to be different per transaction type:

* MetadataAddress probably won't have ``targetId``.
* MetadataMosaic ``targetId`` has UnresovedMosaicId type.
* MetadataNamespace ``targetId`` has a NamespaceId.

I would keep it as it is, but adding different MetadataDTO's per type:

              - $ref: "#/components/schemas/MetadataAddressDTO"
              - $ref: "#/components/schemas/MetadataMosaicDTO"
              - $ref: "#/components/schemas/MetadataNamespaceDTO"

Reply:
yeah anyOf was my initial approach, it's a residue i forgot to remove, thing is, core is only storing one Metadata object for all 3 types, this is why we won't have 3 DTOs  in the end